src/platform/pressureFlowSensor.c

```diff
@@ -0,0 +1,93 @@
+#include "stm32f4xx_hal.h"
+#include <platform/pressureFlowSensor.h>
+#include <platform/pressureFlowSensor_private.h>
+#include <platform/configuration_private.h>
+#include <stdio.h>
+#include <string.h>
+#include <stdint.h>
+
+uint32_t read_FS6122_sensor(uint8_t i2c_addr, uint8_t reg_addr){
+    HAL_StatusTypeDef ret;
+    uint8_t buf[20];
+    uint32_t val; 
+
+    // Tell FS6122 which register to read from (pressure, flow, temp, RH)
+    buf[0] = reg_addr;
+    ret = HAL_I2C_Master_Transmit(&i2c1_bus, i2c_addr, buf, 1, HAL_MAX_DELAY);
+    if ( ret != HAL_OK ) {
+      strcpy((char*)buf, "Error Tx\r\n");
+      val = 128;
```

Comment:
Do not set the return value if there was an error. Should remain zero (or some default value).